dual_screen/android/src/main/kotlin/com/builttoroam/dual_screen/DualScreenPlugin.kt

```diff
@@ -0,0 +1,132 @@
+package com.builttoroam.dual_screen
+
+import android.app.Activity
+import android.content.Context
+import android.view.View
+import android.view.View.OnLayoutChangeListener
+import androidx.annotation.NonNull
+import com.microsoft.device.display.DisplayMask
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.EventChannel
+import io.flutter.plugin.common.EventChannel.EventSink
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+/** DualScreenPlugin */
+public class DualScreenPlugin :
+    FlutterPlugin,
+    MethodCallHandler,
+    EventChannel.StreamHandler,
+    ActivityAware {
+    private val METHOD_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/methods"
+    private val EVENT_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/events"
+    private lateinit var methodChannel: MethodChannel
+    private lateinit var eventChannel: EventChannel
+    private var context: Context? = null
+    private var activity: Activity? = null
+
+    fun registerPlugin(context: Context?, messenger: BinaryMessenger?) {
+        activity = this.activity
```

Comment:
I'm a bit puzzled by this code:

1. What is being assigned to what? Is the this.activity the class activity and the activity is just a local variable?
2. Isn't the context the activity anyways?

Reply:
The activity initialisation wasn't needed here so i have removed that line